attestation-agent/kbs_protocol/src/lib.rs

```diff
@@ -27,55 +27,107 @@ pub trait KbsRequest {
     async fn attest(&mut self, host_url: String) -> Result<String>;
 }
 
+type BoxedAttester = Box<dyn Attester + Send + Sync>;
+
 pub struct KbsProtocolWrapper {
     tee: String,
-    tee_key: Option<TeeKey>,
-    nonce: String,
-    attester: Option<Box<dyn Attester + Send + Sync>>,
+    tee_key: TeeKey,
+    nonce: Option<String>,
+    attester: BoxedAttester,
     http_client: reqwest::Client,
     authenticated: bool,
 }
 
+struct NoTee;
```

Comment:
Why we use generic types here? I saw builder in projects like `reqwest` and `in-toto` uses the following, which seems more straightforward
```rust
pub struct KbsProtocolWrapperBuilder {
    tee: Option<..>,
}

impl KbsProtocolWrapperBuilder {
    pub fn with_tee(mut self, tee: ..) -> Self {
        self.tee = Some(tee);
        self
    } 
}
```

Reply:
Generic types actually will save runtime. I'm ok with both. However there is a naive question from myside. I think the main aim of the PR is to avoid `creating an Attester from the TEE detection heuristic` inside `new()` of `KbsProtocolWrapper`, but to receive an outer `Attester` parameter. If so, why do not directly add a function with the signature

```rust
pub fn new(attester: BoxedAttester) -> Self {
    Self {
        attester,
        ...
    }
}
```

Reply:
> Generic types actually will save runtime. I'm ok with both. However there is a naive question from myside. I think the main aim of the PR is to avoid `creating an Attester from the TEE detection heuristic` inside `new()` of `KbsProtocolWrapper`, but to receive an outer `Attester` parameter. If so, why do not directly add a function with the signature
> 
> ```rust
> pub fn new(attester: BoxedAttester) -> Self {
>     Self {
>         attester,
>         ...
>     }
> }
> ```

you mean `pub fn new_with_attester()` or simply replacing the existing `pub fn new()` function?

if we can do that, the builder pattern is too much. it would maybe be useful if we would want to inject an HTTP client (for unit testing or having a  generic RPC client), but we can also add it later.